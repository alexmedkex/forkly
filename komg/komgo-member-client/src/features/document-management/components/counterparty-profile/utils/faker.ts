import { CounterpartyProfile, RiskLevel } from '../../../../counterparties/store/types'

const defaultProfile: CounterpartyProfile = {
  id: 'anon-profile-id',
  counterpartyId: 'anon-counterparty-id',
  renewalDate: new Date('2019-07-16').toISOString(),
  riskLevel: RiskLevel.unspecified,
  managedById: 'anon-user-id'
}

export const fakeProfile = (partial?: Partial<CounterpartyProfile>) => {
  return partial ? { ...defaultProfile, ...partial } : defaultProfile
}

export const mockDate = () => {
  const RealDate = Date
  const _GLOBAL: any = global // we love typescript!

  return {
    freeze: (isoDate: string | number | Date) => {
      _GLOBAL.Date = class extends RealDate {
        constructor(...args: any) {
          super()
          return new RealDate(isoDate)
        }
      }
    },
    restore: () => {
      global.Date = RealDate
    }
  }
}
