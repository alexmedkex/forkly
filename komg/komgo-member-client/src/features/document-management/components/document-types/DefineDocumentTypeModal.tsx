import * as React from 'react'
import { Form, Modal, Button, Input, Dropdown } from 'semantic-ui-react'
import { Category } from '../../store/types'

export interface Props {
  visible: boolean
  title: string
  categories: Category[]
  predefinedData: { category: string; name: string; id?: string }
  toggleVisible(): void
  onCreateSuccess(cat: string, name: string): void
  onEditSuccess(cat: string, name: string, id: string): void
}

interface State {
  category: string
  name: string
}

interface CategoryRepresentation {
  key: string
}

const formatCategoriesToSelect = (categories: Category[]): CategoryRepresentation[] => {
  return categories.map((cat: Category) => {
    return { key: cat.id, text: cat.name, value: cat.id }
  })
}

class DefineDocumentTypeModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
  }

  handleSubmit = () => {
    if (this.props.predefinedData.id) {
      this.props.onEditSuccess(
        this.state.category ? this.state.category : this.props.predefinedData.category,
        this.state.name,
        this.props.predefinedData.id
      )
    } else {
      this.props.onCreateSuccess(this.state.category, this.state.name)
    }
  }

  render() {
    return (
      <Modal open={this.props.visible} size="mini">
        <Modal.Header>{this.props.title}</Modal.Header>
        <Modal.Content>
          <Modal.Description>
            <Form onSubmit={this.handleSubmit} id="submit-form">
              <Form.Field>
                <label>Category</label>
                <Dropdown
                  placeholder="Select category"
                  defaultValue={
                    this.props.predefinedData.category !== '' ? this.props.predefinedData.category : undefined
                  }
                  options={formatCategoriesToSelect(this.props.categories)}
                  onChange={this.setSelectedCategoryId}
                />
              </Form.Field>
              <Form.Field>
                <label>Name</label>
                <Input
                  placeholder="My new doctype"
                  defaultValue={this.props.predefinedData.name !== '' ? this.props.predefinedData.name : undefined}
                  onChange={(e: any) => {
                    this.setState({ name: e.target.value })
                  }}
                  size="mini"
                />
              </Form.Field>
            </Form>
          </Modal.Description>
        </Modal.Content>
        <Modal.Actions>
          <Button className="ui negative button" role="button" onClick={this.props.toggleVisible}>
            Cancel
          </Button>
          <Button primary={true} content="Submit" form="submit-form" />
        </Modal.Actions>
      </Modal>
    )
  }

  private setSelectedCategoryId = (e: any, { value }: { value: string }) => {
    this.setState({ category: value })
  }
}

export default DefineDocumentTypeModal
