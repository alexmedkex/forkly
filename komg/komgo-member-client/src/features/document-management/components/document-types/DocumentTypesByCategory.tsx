import * as React from 'react'
import { List, Accordion, AccordionPanelProps, Checkbox, Icon } from 'semantic-ui-react'
import styled from 'styled-components'
import { indigo, darkViolet } from '../../../../styles/colors'

import { DocumentType, Category } from '../../../document-management/store/types'
import { groupBy } from '../../../document-management/components/documents/my-documents/toMap'

import { PanelCheckboxState } from '../templates/DocumentRequestModal'

export interface Props {
  documentTypes: DocumentType[]
  categories: Category[]
  selectedDocumentTypesByCategory: Map<string, DocumentType[]>
  panelCheckboxState: Map<string, PanelCheckboxState>
  markAsDisabled: (documentTypeName: string) => { isDisabled: boolean; disabledText: string }
  isSelected(documentType: DocumentType): boolean
  onPanelClick(category: Category): void
  onListItemClick(documentType: DocumentType): void
}

class DocumentTypesByCategory extends React.Component<Props> {
  render() {
    return (
      <StyledAccordion
        exclusive={false}
        fluid={true}
        panels={this.props.categories.map(this.categoryToAccordionPanel)}
      />
    )
  }

  private categoryToAccordionPanel = (category: Category): AccordionPanelProps => {
    return {
      key: category.id,
      name: category.name,
      title: { content: this.categoryToPanelTitle(category) },
      content: { content: this.categoryToList(category) }
    }
  }

  private categoryToPanelTitle = (category: Category) => {
    return (
      <>
        <StyledIcon name="chevron down" />
        <StyledCheckbox
          key={category.id}
          floated="left"
          label={category.name}
          checked={this.props.panelCheckboxState.get(category.id) === 1}
          indeterminate={this.props.panelCheckboxState.get(category.id) === 0}
          onClick={(e: React.MouseEvent) => {
            e.preventDefault()
            e.stopPropagation()
            this.props.onPanelClick(category)
          }}
        />
      </>
    )
  }

  private categoryToList = (category: Category) => {
    const groupByCategoryId = (docType: DocumentType) => docType.category.id
    const documentTypesForThisCategory = groupBy(this.props.documentTypes, groupByCategoryId).get(category.id)
    if (!(documentTypesForThisCategory && documentTypesForThisCategory.length)) {
      return null
    }

    return (
      <List
        selection={true}
        name={<b>category.name</b>}
        items={documentTypesForThisCategory.map(this.documentTypeToListItem)}
      />
    )
  }

  private documentTypeToListItem = (documentType: DocumentType) => {
    // default values
    let disabledText: string = ''
    let isDisabled: boolean = false

    // override defaults if markAsDisabled function is provided
    if (this.props.markAsDisabled) {
      const result = this.props.markAsDisabled(documentType.name)
      if (result.isDisabled) {
        disabledText = result.disabledText
        isDisabled = result.isDisabled
      }
    }
    return (
      <List.Item style={{ padding: '7px 2.5em' }}>
        <List.Content key={documentType.id} floated="left">
          <Checkbox
            disabled={false} // We are allowing to re-request same document several times
            label={documentType.name}
            onClick={() => this.props.onListItemClick(documentType)}
            checked={this.props.isSelected(documentType)}
          />
        </List.Content>
        <List.Content key={documentType.id} floated="right">
          <p className="disabled" hidden={!isDisabled}>
            {disabledText}
          </p>
        </List.Content>
      </List.Item>
    )
  }
}

export const StyledAccordion = styled(Accordion)`
  &&& {
    .komgo-accordion-title {
      background-color: ${indigo} !important;
      border-radius: 5px;
      border: 1px solid ${darkViolet};
      padding: 12px 30px;
      height: 45px;
      color: white;
    }

    .title .dropdown.icon {
      display: none;
    }

    .active .komgo-accordion-title {
      border-radius: 5px 5px 0px 0px;
      margin-bottom: -14px;
    }

    i {
      color: white;
      margin-top: 2px;
    }
  }
`

const StyledCheckbox = styled(Checkbox)`
  &&& {
    padding-left: 1em;
    label {
      font-weight: bold;
    }
    .dropdown.icon {
      display: none;
    }
  }
`

const StyledIcon = styled(Icon)`
  &&& {
    width: 16px;
  }
`

export default DocumentTypesByCategory
