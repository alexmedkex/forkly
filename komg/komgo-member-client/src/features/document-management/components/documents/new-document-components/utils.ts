import { HasId, HasName } from '../../../store/types'

export const hasIdAndNameToOption = (collection: Array<HasId & HasName>): Array<{ key: string }> => {
  const originalCollection = collection.map(el => {
    return { key: el.id, text: el.name, value: el.id }
  })
  return originalCollection.reduce((unique: Array<{ key: string }>, o: { key: string }) => {
    if (!unique.some((obj: { key: string }) => obj.key === o.key)) {
      unique.push(o)
    }
    return unique
  }, [])
}
