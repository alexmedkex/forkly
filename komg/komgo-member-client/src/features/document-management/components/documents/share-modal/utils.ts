import { ICompanyTableItemWithSharedDocs, ICounterpartyWithDocuments } from './SelectedCounterparties'
import { FlagNameValues } from 'semantic-ui-react'
import * as i18nIsoCountries from 'i18n-iso-countries'

export function pluralize(word: string, count: number, singleSuffix: string, pluralSuffix: string) {
  if (count > 1) {
    return `${word}${pluralSuffix}`
  }

  return `${word}${singleSuffix}`
}

export const buildCounterpartyWithDocumentsPickerItems = (
  counterpartiesWithDocs: ICounterpartyWithDocuments[]
): ICompanyTableItemWithSharedDocs[] => {
  i18nIsoCountries.registerLocale(require('i18n-iso-countries/langs/en.json'))
  return counterpartiesWithDocs.map(item => ({
    name: item.counterparty.x500Name.CN,
    countryName: i18nIsoCountries.getName(item.counterparty.x500Name.C.trim(), 'en'),
    country: item.counterparty.x500Name.C.toLocaleLowerCase().trim() as FlagNameValues,
    location: item.counterparty.x500Name.L,
    id: item.counterparty.staticId,
    sharedDocuments: item.documents
  }))
}
