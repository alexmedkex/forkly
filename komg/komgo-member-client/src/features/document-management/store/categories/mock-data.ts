import { Category } from '../types'
import { mockProduct } from '../products/mock-data'

export const mockCategories: Category[] = [
  {
    id: 'company-details',
    product: mockProduct,
    name: 'Company Details'
  },
  {
    id: 'management-and-directors',
    product: mockProduct,
    name: 'Management and directors'
  },
  {
    id: 'shareholders',
    product: mockProduct,
    name: 'Shareholders - Ultimate Beneficiary Owners (UBOs)'
  },
  {
    id: 'business-description',
    product: mockProduct,
    name: 'Business description'
  },
  {
    id: 'regulation-and-compliance',
    product: mockProduct,
    name: 'Regulation/Compliance'
  },
  {
    id: 'banking-documents',
    product: mockProduct,
    name: 'Banking documents'
  },
  {
    id: 'miscellaneous',
    product: mockProduct,
    name: 'Miscellaneous'
  }
]
