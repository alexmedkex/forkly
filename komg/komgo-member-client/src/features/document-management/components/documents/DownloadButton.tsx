import { Button } from 'semantic-ui-react'
import { Document, DocumentType } from '../../store'
import CommentTooltip from '../../../review-documents/components/CommentTooltip'
import { CustomInformationIcon } from '../../../../components/custom-icon/InformationIcon'
import React from 'react'
import { initiateDownload } from '../../utils/downloadDocument'
import styled from 'styled-components'
import DocumentViewContainer, { HeaderActions } from '../../containers/DocumentViewContainer'
import DocumentSimpleHeader from './DocumentSimpleHeader'
import DocumentSimpleInfo from './DocumentSimpleInfo'

interface Props {
  document: Document
  docType: DocumentType
  onOriginalDocument(doc: Document): void
  onDocumentDownload?(doc: Document): void
}

interface State {
  isDownloaded: boolean
  viewDocumentId: string
}

export class DownloadButton extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = { isDownloaded: this.isAlreadyDownloaded(this.props.document), viewDocumentId: '' }
  }

  handleViewDocument = (id: string) => {
    this.setState({ viewDocumentId: id })
  }

  handleCloseViewModal = () => {
    this.setState({ viewDocumentId: '' })
  }

  render() {
    if (this.state.viewDocumentId) {
      return (
        <DocumentViewContainer
          documentId={this.state.viewDocumentId}
          onClose={this.handleViewDocument}
          renderHeader={(document: Document, actions: HeaderActions) => (
            <DocumentSimpleHeader document={document} actions={actions} />
          )}
          renderInfoSection={(document: Document) => (
            <DocumentSimpleInfo document={{ ...document, registrationDate: undefined }} />
          )}
        />
      )
    }
    if (!this.isAlreadyDownloaded(this.props.document) && !this.state.isDownloaded) {
      return (
        <StyledDownloadButton>
          <Button
            data-test-id={`requirements-add-button-${this.props.docType.name}`}
            content="Download attachment"
            style={{
              width: '153px',
              fontSize: '14px',
              height: '28px',
              margin: '0 10px 0 0',
              padding: '0',
              textRendering: 'geometricPrecision'
            }}
            default={true}
            disabled={false}
            primary={true}
            onClick={() => {
              initiateDownload(this.props.document)
              if (this.props.onDocumentDownload) {
                this.props.onDocumentDownload(this.props.document)
              }
              this.setState({
                isDownloaded: true
              })
            }}
          />

          <CommentTooltip
            comment={`The counterparty has attached a document for you to use. You must first download it in order to proceed`}
            icon={
              <CustomInformationIcon
                size="large"
                name="comment alternate"
                style={{ marginLeft: '0px', fontSize: '1.2em', alignSelf: 'center' }}
              />
            }
          />
        </StyledDownloadButton>
      )
    } else {
      return (
        <Button
          data-test-id={`requirements-add-button-${this.props.docType.name}`}
          content="Original attachment"
          style={{
            width: '153px',
            fontSize: '14px',
            height: '28px',
            margin: '0 10px 0 0',
            padding: '0'
          }}
          default={true}
          disabled={false}
          onClick={() => this.handleViewDocument(this.props.document.id)}
        />
      )
    }
  }

  private isAlreadyDownloaded(doc: Document) {
    if (doc && doc.downloadInfo) {
      const downloadedByUsers = doc.downloadInfo.downloadedByUsers
      return !!downloadedByUsers && downloadedByUsers.length > 0
    }
    return false
  }
}

const StyledDownloadButton = styled.div`
  &&&&&&&&& {
    display: flex;
  }
`
