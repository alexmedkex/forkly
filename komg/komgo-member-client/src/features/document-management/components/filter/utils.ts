import { Category, DocumentType } from '../../index'
import { sortCategories, sortDocumentTypes } from '../../utils/sortingHelper'
import { categoryToColor } from '../documents/document-library/categoryToColor'

export const getDocumentTypeFilterOptions = (categories: Category[], types: DocumentType[]) => {
  return sortCategories(categories).map(c => ({
    name: c.name,
    value: c.id,
    display: {
      color: categoryToColor[c.id]
    },
    items: sortDocumentTypes(types)
      .filter(t => t.category.id === c.id)
      .map(t => ({
        name: t.name,
        value: t.id
      }))
  }))
}
