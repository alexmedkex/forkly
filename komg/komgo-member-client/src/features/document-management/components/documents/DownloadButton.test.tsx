import { shallow, mount } from 'enzyme'
import * as React from 'react'
import * as util from 'util'
import { Button } from 'semantic-ui-react'
import { DownloadButton } from './DownloadButton'
import { fakeDocument } from '../../utils/faker'
import { DocumentType } from '../../store'
import { mockDocumentTypes } from '../../store/document-types/mock-data'
import * as renderer from 'react-test-renderer'

describe('DownloadButton component', () => {
  const mockFunc = jest.fn(() => void 0)
  const mockDocument = fakeDocument()
  const mockDocType: DocumentType = mockDocumentTypes[0]

  it('renders', () => {
    const props = {
      document: mockDocument,
      docType: mockDocType,
      onOriginalDocument: mockFunc,
      onDocumentDownload: mockFunc
    }

    expect(renderer.create(<DownloadButton {...props} />).toJSON()).toMatchSnapshot()
  })
})
