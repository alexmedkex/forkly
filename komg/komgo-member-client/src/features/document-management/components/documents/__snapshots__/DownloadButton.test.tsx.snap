// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadButton component renders 1`] = `
.c0.c0.c0.c0.c0.c0.c0.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
>
  <button
    className="ui primary button"
    data-test-id="requirements-add-button-Certificate of Incorporation"
    default={true}
    onClick={[Function]}
    role="button"
    style={
      Object {
        "fontSize": "14px",
        "height": "28px",
        "margin": "0 10px 0 0",
        "padding": "0",
        "textRendering": "geometricPrecision",
        "width": "153px",
      }
    }
  >
    Download attachment
  </button>
  <svg
    name="comment alternate"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="large"
    style={
      Object {
        "alignSelf": "center",
        "fontSize": "1.2em",
        "marginLeft": "0px",
      }
    }
  >
    information-outline.svg
  </svg>
</div>
`;
